### PR TITLE
Fix Rust-Analyzer artifact URL

### LINKED ISSUE #373: Stuck at "Downloading rust-analyzer... {0}%"
Hello!

I have installed Corrosion: Rust edition in Eclipse IDE 1.2.0.202103130942 installed on Eclipse 2021-03 (4.19.0).

Whenever I click on "Download rust-analyzer" it gets stuck on "Downloaing rust-analyzer... {0}%" without no progress. I also see "Input stream must be configured" and "Unhandled event loop exception" error messages in error log.

Thanks!

### PR DESCRIPTION
**Pull request by rootkea**
Closes #373

### CODE REVIEW

**Comment by rootkea** [comment]
I have never programmed in Java so literally have no idea how to test this change I made.

Also shouldn't there be `".gz"` string appended at the end of URL?

**Comment by rootkea** [comment]
And what about `rust-analyzer-x86_64-unknown-linux-musl.gz`? When to use that artifact?

File: org.eclipse.corrosion/src/org/eclipse/corrosion/RustManager.java [hunk 1]
```diff
@@ -313,9 +313,11 @@ public static File getLanguageServerExecutable() {
...
 		}
 		String filename = "rust-analyzer-" + //$NON-NLS-1$
-				(Platform.OS_LINUX.equals(Platform.getOS()) ? "linux" : //$NON-NLS-1$
-						Platform.OS_WIN32.equals(Platform.getOS()) ? "win.exe" : //$NON-NLS-1$
-								Platform.OS_MACOSX.equals(Platform.getOS()) ? "mac" : "os-not-found"); //$NON-NLS-1$ //$NON-NLS-2$
+				("aarch64".equals(Platform.getOSArch()) ? "aarch64-" : //$NON-NLS-1$ //$NON-NLS-2$
+						"x86_64".equals(Platform.getOSArch()) ? "x86_64-" : "arch-not-found") //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+				+ (Platform.OS_LINUX.equals(Platform.getOS()) ? "unknown-linux-gnu" : //$NON-NLS-1$
+						Platform.OS_WIN32.equals(Platform.getOS()) ? "pc-windows-msvc" : //$NON-NLS-1$
+								Platform.OS_MACOSX.equals(Platform.getOS()) ? "apple-darwin" : "os-not-found"); //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (mickaelistria) [inline org.eclipse.corrosion/src/org/eclipse/corrosion/RustManager.java:320]:
```suggestion
				(Platform.ARCH_AARCH64.equals(Platform.getOSArch()) ? "aarch64-" : //$NON-NLS-1$ //$NON-NLS-2$
						Platform.ARCH_X86_64.equals(Platform.getOSArch()) ? "x86_64-" : "arch-not-found") //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
				+ (Platform.OS_LINUX.equals(Platform.getOS()) ? "unknown-linux-gnu" : //$NON-NLS-1$
						Platform.OS_WIN32.equals(Platform.getOS()) ? "pc-windows-msvc" : //$NON-NLS-1$
								Platform.OS_MACOSX.equals(Platform.getOS()) ? "apple-darwin" : "os-not-found"); //$NON-NLS-1$ //$NON-NLS-2$
```

**Review by mickaelistria — CHANGES_REQUESTED** [review]
Indeed, it seem s like .gz needs to be appended and an extra step added to gunzip the distribution. Do you think you can take care of this?

**Comment by rootkea** [comment]
Googlefoo says I gotta use `GZIPInputStream` but I'm afraid this will take more time as Java being unknown waters for me. So maybe you should go ahead and add the .gz decompress part. :)

**Pull request closed without merging**